ports/geckolib/glue.rs

```diff
@@ -267,12 +268,68 @@ pub extern "C" fn Servo_GetComputedValuesForAnonymousBox(parent_style_or_null: *
     type Helpers = ArcHelpers<ServoComputedValues, GeckoComputedValues>;
 
     Helpers::maybe_with(parent_style_or_null, |maybe_parent| {
-        let new_computed = data.stylist.computed_values_for_pseudo(&pseudo, maybe_parent);
+        let new_computed = data.stylist.precomputed_values_for_pseudo(&pseudo, maybe_parent);
         new_computed.map_or(ptr::null_mut(), |c| Helpers::from(c))
     })
 }
 
 #[no_mangle]
+pub extern "C" fn Servo_GetComputedValuesForPseudoElement(parent_style: *mut ServoComputedValues,
+                                                          match_element: *mut RawGeckoElement,
+                                                          pseudo_tag: *mut nsIAtom,
+                                                          raw_data: *mut RawServoStyleSet,
+                                                          is_probe: bool)
+     -> *mut ServoComputedValues {
+    debug_assert!(!match_element.is_null());
+
+    let parent_or_null = || {
+        if is_probe {
+            ptr::null_mut()
+        } else {
+            Servo_AddRefComputedValues(parent_style);
+            parent_style
+        }
+    };
+
+    let pseudo = match pseudo_element_from_atom(pseudo_tag, /* ua_stylesheet = */ true) {
```

Comment:
We'll want to pass the appropriate value here from the caller right? I'll add a TODO.